### PR TITLE
Customizer: Fix Menus Redirect for unverified E-Mail

### LINKED ISSUE #13226: Visiting the Customizer with an Unverified Email address errors out
## The error
screen-shot-2017-04-18-at-17-27-17

`Uncaught RangeError: Maximum call stack size exceeded`

## Steps to reproduce
* Create a new WordPress.com account and site via `/start` and **DO NOT** verify your email address via the link sent to you in email
* Open `Calypso | Sites` & make sure your new site is selected
* Click `Customize` in the Sidebar (or from the header on the Themes page -- or just browse to `/customize/%slug%`

@bernhardreiter, I think this issue is related to this change #13082.

If I short-circuit the `Customize.redirectIfNeeded` function, the error does not occur.

Reported @ p58i-5gt-p2#comment-35398

### PR DESCRIPTION
**Pull request by ockham**
There's a conceptual error in my original check -- comparing `getMenusUrl` to `/customize/menus/ + site.slug` doesn't cut it, if we really are at `/customize/ + site.slug`.

Instead of building that RHS string ourselves, we should compare to `context.pathname` -- which is what we're really meaning: compare to current path, and if it's not what `getMenusUrl` gives us, redirect to that.

This way, we keep leveraging `getMenusUrl`, thus mirroring the sidebar.

To test:
* Land on `http://calypso.localhost:3000/menus` and `http://calypso.localhost:3000/menus/<yoursite>`
* Verify that you're being redirected to `http://calypso.localhost:3000/customize/menus` and `http://calypso.localhost:3000/cusotmize/menus/<yoursite>`, respectively
* Use a JP site for `<yoursite>`. Verify you're redirected to that site's customizer.
* With a user with unverified email, check that you're redirected to `/customize/<yoursite>` (no `/menus` route fragment)
* Also make sure that the regular 'Customize' menu item still takes you to the Customizer as before.

Fixes #13226

cc @kwight @mattwiebe

### CODE REVIEW

**Comment by ockham** [comment]
Hmm, needs some more polish, since it also redirects non-menus customizer routes now...

File: client/my-sites/customize/main.jsx [hunk 1]
```diff
@@ -65,11 +66,12 @@ var Customize = React.createClass( {
...
-		this.redirectIfNeeded( nextProps.menusUrl, nextProps.site );
+		this.redirectIfNeeded( nextProps.pathname );
 	},
 
-	redirectIfNeeded: function( menusUrl, site ) {
-		if ( site && menusUrl !== '/customize/menus/' + site.slug ) {
+	redirectIfNeeded: function( pathname ) {
+		const { menusUrl } = this.props;
+		if ( startsWith( pathname, '/customize/menus' ) && pathname !== menusUrl ) {
 			page( menusUrl );
```
Comment (ockham) [inline client/my-sites/customize/main.jsx]: Oh, we might actually want to use [`page.redirect`](https://github.com/visionmedia/page.js#pageredirectpath) here. That leaves a cleaner Browser history...

Comment (ockham) [inline client/my-sites/customize/main.jsx]: Ah, nvm, we aren't allowed to do that. Different origins!

**Comment by kwight** [comment]
Tested with unverified and verified emails, both menus and `customize/widgets` routes – all good :shipit:

**Comment by jblz** [comment]
Thanks for the fix!

**Pull request merged**